pinot-core/src/main/java/org/apache/pinot/core/query/pruner/ColumnValueSegmentPruner.java

```diff
@@ -19,184 +19,224 @@
 package org.apache.pinot.core.query.pruner;
 
 import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
-import javax.annotation.Nonnull;
+import java.util.Set;
 import org.apache.commons.configuration.Configuration;
-import org.apache.pinot.spi.data.FieldSpec;
-import org.apache.pinot.common.request.FilterOperator;
+import org.apache.pinot.common.request.transform.TransformExpressionTree;
 import org.apache.pinot.common.utils.request.FilterQueryTree;
+import org.apache.pinot.core.common.DataSource;
+import org.apache.pinot.core.common.DataSourceMetadata;
 import org.apache.pinot.core.common.predicate.RangePredicate;
+import org.apache.pinot.core.data.partition.PartitionFunction;
 import org.apache.pinot.core.indexsegment.IndexSegment;
+import org.apache.pinot.core.query.exception.BadQueryRequestException;
 import org.apache.pinot.core.query.request.ServerQueryRequest;
-import org.apache.pinot.core.segment.index.ColumnMetadata;
-import org.apache.pinot.core.segment.index.SegmentMetadataImpl;
 import org.apache.pinot.core.segment.index.readers.BloomFilterReader;
+import org.apache.pinot.spi.data.FieldSpec.DataType;
+import org.apache.pinot.spi.utils.BytesUtils;
 
 
 /**
- * An implementation of SegmentPruner.
- * <p>Pruner will prune segment based on the column value inside the filter.
+ * The {@code ColumnValueSegmentPruner} is the segment pruner that prunes segments based on the value inside the filter.
```

Comment:
It would be nice if this javadoc can be enhanced to highlight what all pruning is done. Something like..

(1) use equality filter value for partition pruning (if table is partitioned)
(2) use equality filter for bloom filter based pruning
(2) min-max value based pruner for RANGE

Reply:
Updated the doc